digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/pointasset/TrafficSignService.scala

```diff
@@ -152,7 +156,23 @@ class TrafficSignService(val roadLinkService: RoadLinkService, val userProvider:
 
   override def getByBoundingBox(user: User, bounds: BoundingRectangle): Seq[PersistedAsset] = {
     val (roadLinks, changeInfo) = roadLinkService.getRoadLinksWithComplementaryAndChangesFromVVH(bounds)
-    super.getByBoundingBox(user, bounds, roadLinks, changeInfo, floatingAdjustment(adjustmentOperation, createOperation))
+    val result = super.getByBoundingBox(user, bounds, roadLinks, changeInfo, floatingAdjustment(adjustmentOperation, createOperation))
+    val (pc, others) = result.partition(asset => asset.createdBy.contains(batchProcessName) && asset.propertyData.find(_.publicId == typePublicId).get.values.head.propertyValue.toInt == TrafficSignType.PedestrianCrossing.value)
+
+    sortCrossings(pc, Seq()) ++ others
+  }
+
+  def sortCrossings(sorted: Seq[PersistedAsset], result: Seq[PersistedAsset]): Seq[PersistedAsset] = {
+    val centerSignOpt = sorted.headOption
```

Comment:
Why are we getting the headOption, this could lead to a mistake, if we have assets with latitude as 21, 20 and 18, and if we pick the headOption, and is the one with latitude 21 the asset with latitude 18 is excluded. And if we get the middle one the ( one with latitude 20 ) we get all together.